src/Infiltration/ui/SlashGame.tsx

```diff
@@ -25,11 +25,30 @@ const difficulties: {
   Impossible: { window: 150 },
 };
 
-export function SlashGame({ difficulty: _difficulty, onSuccess, onFailure }: IMinigameProps): React.ReactElement {
-  const difficulty: Difficulty = { window: 0 };
-  interpolate(difficulties, _difficulty, difficulty);
-
+export function SlashGame({ difficulty, onSuccess, onFailure }: IMinigameProps): React.ReactElement {
   const [phase, setPhase] = useState(0);
+  const [hasAugment, setHasAugment] = useState(false);
+  const [guardingTime, setGuardingTime] = useState(0);
+
+  useEffect(() => {
+    const newDifficulty: Difficulty = { window: 0 };
+    interpolate(difficulties, difficulty, newDifficulty);
+    const timePreparing = newDifficulty.window;
+    const timeAttacking = 250;
+    const timeGuarding = Math.random() * 3250 + 1500 - (timeAttacking + timePreparing);
+    setGuardingTime(timeGuarding);
+    setHasAugment(Player.hasAugmentation(AugmentationName.MightOfAres, true));
+
+    let id = setTimeout(() => {
+      setPhase(1);
+      id = setTimeout(() => {
+        setPhase(2);
+        id = setTimeout(() => onFailure(), timeAttacking);
+      }, timePreparing);
+    }, timeGuarding);
+
+    return () => clearTimeout(id);
+  }, [difficulty, onSuccess, onFailure]);
```

Comment:
So when we added variables to the dependency array of the useEffect that initializes the game timer, we are basically saying that changing props on the component after it's already is a reinitialization of the game.

However, while we made the changes in the useEffect so that prop changes will reinitialize the game timer, other calculated variables were still set up so that they are only initialized once per mount. So the game timer initializes once per props update, but the relevant calculated values only recalculate on unmount/remount, the component is really not able to fully reinitialize the game on props update.

So what I've done here is move the initializers for these calculated values into the useEffect, so they will all be recalculated on a props change. This means that, even though infiltration is currently written in a way that it will always unmount/remount for each game because of the timer displayed between games, this SlashGame component is robust enough that it can actually handle being displayed twice in a row without needing a fresh mount to initialize a new game.

(Actually I need to setPhase back to 0 during the useEffect as well, will add another commit for that)